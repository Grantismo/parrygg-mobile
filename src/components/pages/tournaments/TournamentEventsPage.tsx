import React, { useState } from "react";
import {
  GestureResponderEvent,
  TouchableOpacity,
  View,
  Image,
} from "react-native";
import DraggableFlatList, {
  RenderItemParams,
  ShadowDecorator,
} from "react-native-draggable-flatlist";

import Background from "@/components/base/Background";
import Button from "@/components/base/Button";
import Text from "@/components/base/Text";
import Nav from "@/components/base/navigation/Nav";
import SettingsButton from "@/components/base/settings/SettingsButton";
import TournamentBreadcrumb from "@/components/base/tournaments/TournamentBreadcrumb";
import tw from "@/lib/tailwind";
import DragHandle from "@assets/icons/DragHandle";
import Plus from "@assets/icons/Plus";

interface Event {
  title: string;
  imgSrc: string;
}

interface EventButtonParams {
  title: string;
  imgSrc: string;
  dragProps: {
    drag: (event: GestureResponderEvent) => void;
    disabled: boolean;
  };
}
const EventButton = ({
  title,
  imgSrc,
  dragProps: { drag, disabled },
}: EventButtonParams) => {
  return (
    <SettingsButton path="">
      <View style={tw`w-full flex flex-row items-center justify-between`}>
        <View style={tw`flex-row items-center justify-center`}>
          <TouchableOpacity hitSlop={10} onPressIn={drag} disabled={disabled}>
            <DragHandle style={tw`mr-3`} color="#777777" />
          </TouchableOpacity>
          <Image style={tw`h-6 w-10 mr-4 rounded-2xs`} src={imgSrc} />
          <Text>{title}</Text>
        </View>
      </View>
    </SettingsButton>
  );
};

const TournamentEventsPage = () => {
  const [events, setEvents] = useState<Event[]>([
    { title: "SSBU: Singles", imgSrc: "https://i.imgur.com/I3HiTD7.png" },
    { title: "SSBU: Doubles", imgSrc: "https://i.imgur.com/ETSVJpG.png" },
    { title: "SF6: Singles", imgSrc: "https://i.imgur.com/I3HiTD7.png" },
  ]);

  const renderItem = ({ item, drag, isActive }: RenderItemParams<Event>) => {
    return (
      <ShadowDecorator>
        <EventButton {...item} dragProps={{ drag, disabled: isActive }} />
      </ShadowDecorator>
    );
  };
  return (
    <Background style={tw`p-0 justify-start`}>
      <Nav title="Manage Events" showBack collapsed />
      <TournamentBreadcrumb />
      <View style={tw`h-full w-full p-6`}>
        <DraggableFlatList
          data={events}
          onDragEnd={({ data }) => setEvents(data)}
          keyExtractor={(s) => s.title}
          renderItem={renderItem}
        />
        <Button size="lg" style={tw`w-full`}>
          <Plus color="#1B1B1B" style={tw`mr-2`} />
          <Text color="black">Add Event</Text>
        </Button>
      </View>
    </Background>
  );
};

export default TournamentEventsPage;
