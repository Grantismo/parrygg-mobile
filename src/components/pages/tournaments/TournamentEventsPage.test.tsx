import { describe, expect, it, jest } from "@jest/globals";
import React from "react";
import renderer from "react-test-renderer";
import "react-native-draggable-flatlist";

import TournamentEventsPage from "@/components/pages/tournaments/TournamentEventsPage";

// Flatlist causes a recursive snapshot error so we mock it out
jest.mock("react-native-draggable-flatlist", () => {
  return {
    __esModule: true,
    default: jest.fn(),
  };
});

describe("CreateTournamentPage", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<TournamentEventsPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
