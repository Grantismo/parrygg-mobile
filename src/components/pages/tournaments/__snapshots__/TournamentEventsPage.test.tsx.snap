// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTournamentPage renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#1b1b1b",
    }
  }
>
  <View
    style={
      {
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "opacity": 0.03,
          "resizeMode": "cover",
        }
      }
    />
  </View>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      [
        {
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
        },
      ]
    }
  >
    <View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "rgba(10, 10, 10, 1)",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 20,
            "paddingLeft": 32,
            "paddingRight": 32,
            "paddingTop": 14,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "width": 56,
            }
          }
        />
        <Text
          collapsable={false}
          style={
            {
              "color": "#fff",
              "fontFamily": "Inter_600SemiBold",
              "fontSize": 16,
            }
          }
        >
          Manage Events
        </Text>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "width": 56,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={5}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "Inter_500Medium",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "color": "#777",
                  },
                  undefined,
                ]
              }
            >
              Back
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "height": "100%",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
            [
              {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "center",
              },
              {
                "justifyContent": "flex-start",
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ],
          ]
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          colors={
            [
              4280756007,
              4280427042,
            ]
          }
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 16,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 16,
              "width": "100%",
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="15"
            bbWidth="14"
            fill="none"
            focusable={false}
            height={15}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "marginRight": 8,
                },
                {
                  "flex": 0,
                  "height": 15,
                  "width": 14,
                },
              ]
            }
            vbHeight={15}
            vbWidth={14}
            width={14}
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M8.40741 0.654687C8.11228 0.40595 7.73873 0.269531 7.35276 0.269531C6.96679 0.269531 6.59324 0.40595 6.29811 0.654687L1.38852 4.79693C1.20629 4.95051 1.0598 5.14204 0.959294 5.35813C0.858788 5.57421 0.80669 5.80964 0.806641 6.04796V12.8159C0.806641 13.5185 1.37688 14.0888 2.0795 14.0888H4.26154C4.59912 14.0888 4.92288 13.9547 5.16158 13.716C5.40029 13.4773 5.53439 13.1535 5.53439 12.8159V9.90435C5.53439 9.40976 5.92862 9.00826 6.41957 8.99517H8.28595C8.52289 9.00143 8.74803 9.09997 8.91337 9.2698C9.07872 9.43964 9.17121 9.66733 9.17113 9.90435V12.8159C9.17113 13.5185 9.74137 14.0888 10.444 14.0888H12.626C12.9636 14.0888 13.2874 13.9547 13.5261 13.716C13.7648 13.4773 13.8989 13.1535 13.8989 12.8159V6.04724C13.8988 5.80892 13.8467 5.57349 13.7462 5.3574C13.6457 5.14131 13.4992 4.94978 13.317 4.7962L8.40741 0.654687Z"
                fill={
                  {
                    "payload": 4294967295,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <Text
            style={
              [
                {
                  "fontFamily": "Inter_500Medium",
                },
                {
                  "fontSize": 16,
                },
                {
                  "color": "#777",
                },
                {
                  "marginRight": 8,
                },
              ]
            }
          >
            \\
          </Text>
          <Text
            style={
              [
                {
                  "fontFamily": "Inter_500Medium",
                },
                {
                  "fontSize": 16,
                },
                {
                  "color": "#FFC93F",
                },
                {
                  "marginRight": 8,
                },
              ]
            }
          >
            Manage Events
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
        <View
          style={
            {
              "height": "100%",
              "paddingBottom": 24,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 24,
              "width": "100%",
            }
          }
        >
          <View
            style={
              [
                {
                  "marginBottom": 16,
                },
                {
                  "width": "100%",
                },
              ]
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              colors={
                [
                  4293899034,
                  4294953798,
                ]
              }
              style={
                [
                  {
                    "backgroundColor": "rgba(56, 52, 42, 0.38)",
                  },
                  {
                    "borderRadius": 14,
                  },
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderWidth": 2,
                      "color": "#000",
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingBottom": 15,
                      "paddingLeft": 24,
                      "paddingRight": 24,
                      "paddingTop": 13,
                    },
                    {
                      "borderColor": "#ffd978",
                    },
                    false,
                    false,
                    false,
                    false,
                    {
                      "borderRadius": 14,
                    },
                    undefined,
                  ]
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="18"
                  bbWidth="18"
                  fill="none"
                  focusable={false}
                  height={18}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "marginRight": 8,
                      },
                      {
                        "flex": 0,
                        "height": 18,
                        "width": 18,
                      },
                    ]
                  }
                  vbHeight={18}
                  vbWidth={18}
                  width={18}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M8.99809 0.0605469C4.06081 0.0605469 0.0585938 4.06276 0.0585938 9.00005C0.0585938 13.9373 4.06081 17.9395 8.99809 17.9395C13.9354 17.9395 17.9376 13.9373 17.9376 9.00005C17.9376 4.06276 13.9354 0.0605469 8.99809 0.0605469ZM13.4678 9.894H9.89204V13.4698H8.10414V9.894H4.52834V8.1061H8.10414V4.5303H9.89204V8.1061H13.4678V9.894Z"
                      fill={
                        {
                          "payload": 4279966491,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Inter_500Medium",
                      },
                      {
                        "fontSize": 16,
                      },
                      {
                        "color": "#000",
                      },
                      undefined,
                    ]
                  }
                >
                  Add Event
                </Text>
              </View>
            </ViewManagerAdapter_ExpoLinearGradient>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
